### PR TITLE
feat(restingOrder): Upgrade to new mangrove-core supporting expiry as a timestamp

### PR DESCRIPTION
**Pull request by lnist**
… a timestamp

### CODE REVIEW

File: packages/mangrove.js/src/market.ts [hunk 1]
```diff
@@ -758,30 +757,37 @@ class Market {
...
 
-      txPromise.then((resp) => {
-        // Warning: if the tx nor any with the same nonce is ever mined,
-        // the `once` and block callbacks will never be triggered and you will memory leak by queuing tasks.
-        txHashDeferred.resolve(resp.hash);
-        resp
-          .wait(1)
-          .then((recp) => {
-            this.afterBlock(recp.blockNumber, async () => {
+      txPromise
```
Comment (adhusson) [inline packages/mangrove.js/src/market.ts:760]: the prettification makes it difficult to understand what has changed

File: packages/mangrove.js/src/market.ts [hunk 2]
```diff
@@ -758,30 +757,37 @@ class Market {
...
             });
-          })
-          .catch((e) => {
-            this.unsubscribe(cb);
-            ko({ revert: true, exception: e });
-          });
-      });
+        })
+        .catch((e) => {
+          txHashDeferred.reject(e);
```
Comment (adhusson) [inline packages/mangrove.js/src/market.ts:787]: yes seems good! agreed that this is just cleaning up our objects and there is no behavior change, right?

File: packages/mangrove.js/src/offerLogic.ts [hunk 3]
```diff
@@ -154,6 +154,7 @@ class OfferLogic {
   ): Promise<Big> {
     const rawBalance = await this.contract.tokenBalance(
       this.mgv.getAddress(tokenName),
+      await this.mgv._signer.getAddress(),
```
Comment (adhusson) [inline packages/mangrove.js/src/offerLogic.ts:157]: not sure what this is for

File: packages/mangrove.js/test/unit/trade.unit.test.ts [hunk 4]
```diff
@@ -52,7 +52,9 @@ describe("Trade unit tests suite", () => {
       const expectedGivesWithoutSlippage = Big(params.volume).mul(price);
       assert.equal(result.wants.eq(BigNumber.from(params.volume)), true);
       assert.equal(
-        result.givesWithoutSlippage.eq(expectedGivesWithoutSlippage),
+        BigNumber.from(expectedGivesWithoutSlippage.toFixed(0)).eq(
```
Comment (adhusson) [inline packages/mangrove.js/test/unit/trade.unit.test.ts:55]: why did the equality test change from
`a.eq(b)` to `BigNumber.from(b.toFixed(0)).eq(a'))` ?

File: packages/mangrove.js/src/offerLogic.ts (hunk 3 above)
Comment (lnist) [reply packages/mangrove.js/src/offerLogic.ts:157]: the tokenBalance now takes a maker as a parameter.

File: packages/mangrove.js/src/market.ts (hunk 1 above)
Comment (lnist) [reply packages/mangrove.js/src/market.ts:760]: yeah - I use kdiff3 to get something meaningful. See slack :)

File: packages/mangrove.js/src/offerLogic.ts (hunk 3 above)
Comment (jkrivine) [reply packages/mangrove.js/src/offerLogic.ts:157]: I think tokenBalance should have an optional `maker` parameter. If none are provided then `this.mgv._signer.getAddress()` should be used

**Review by jkrivine — CHANGES_REQUESTED** [review]
Looks good to me except maybe a minor change in `tokenBalance`

File: packages/mangrove.js/src/market.ts [hunk 5]
```diff
@@ -758,30 +755,43 @@ class Market {
...
+                    revert: false,
+                    exception: "tx mined but filter never returned true",
+                  });
+                }
+              })
+                .then(() => {
+                  // ignored
+                })
+                .catch((e) => {
+                  throw e;
```
Comment (adhusson) [inline packages/mangrove.js/src/market.ts:781]: I think it's actually fine without it... sorry!

Comment (lnist) [reply packages/mangrove.js/src/market.ts]: removed :)

**Review by jkrivine — APPROVED** [review]

**Pull request merged**